tests/test_cmd_resetdep.py

```diff
@@ -0,0 +1,98 @@
+        assert get_file_md5(get_abspath("data/dependency1")) == md5
+
+    def test_file_dep_up_to_date(self, depfile, dependency1):
+        my_task = Task("t2", [""], file_dep=['tests/data/dependency1'])
+        depfile.save_success(my_task)
+        output = StringIO()
+        cmd_list = CmdFactory(ResetDep, outstream=output, task_list=[my_task],
+                              dep_manager=depfile)
+        cmd_list._execute()
+        got = output.getvalue()
+        assert "skip t2\n" == got
+
+    def test_file_dep_change_checker(self, depfile, dependency1):
+        my_task = Task("t2", [""], file_dep=['tests/data/dependency1'])
+        depfile.save_success(my_task)
+        depfile.checker = TimestampChecker()
+        output = StringIO()
+        cmd_list = CmdFactory(ResetDep, outstream=output, task_list=[my_task],
+                              dep_manager=depfile)
+        cmd_list._execute()
+        got = output.getvalue()
+        assert "processed t2\n" == got
+
+    def test_filter(self, depfile, dependency1):
+        output = StringIO()
+        tasks = tasks_sample()
+        cmd_list = CmdFactory(ResetDep, outstream=output, task_list=tasks,
+                              dep_manager=depfile)
+        cmd_list._execute(pos_args=['t2'])
+        got = output.getvalue()
+        assert "processed t2\n" == got
+
+    def test_invalid_task(self, depfile):
+        output = StringIO()
+        tasks = tasks_sample()
+        cmd_list = CmdFactory(ResetDep, outstream=output, task_list=tasks,
+                              dep_manager=depfile)
+        pytest.raises(InvalidCommand, cmd_list._execute, pos_args=['xxx'])
+
+    def test_missing_file_dep(self, depfile):
+        my_task = Task("t2", [""], file_dep=['tests/data/missing'])
+        output = StringIO()
+        cmd_list = CmdFactory(ResetDep, outstream=output, task_list=[my_task],
+                              dep_manager=depfile)
+        cmd_list._execute()
+        got = output.getvalue()
+        assert ("failed t2 (missing file_dep: [Errno 2] No such file or "
+                "directory: 'tests/data/missing')\n") == got
+
+    def test_values_and_results(self, depfile, dependency1):
+        my_task = Task("t2", [""], file_dep=['tests/data/dependency1'])
+        my_task.result = "result"
+        my_task.values = {'x': 5, 'y': 10}
+        depfile.save_success(my_task)
+        depfile.checker = TimestampChecker()  # trigger task update
+        output = StringIO()
+        cmd_list = CmdFactory(ResetDep, outstream=output, task_list=[my_task],
+                              dep_manager=depfile)
+        cmd_list._execute()
+        got = output.getvalue()
+        # ipdb> from doit.dependency import DbmDependency
+        # ipdb> db = DbmDependency(depfile.name)
+        assert "processed t2\n" == got
+        assert {'x': 5, 'y': 10} == depfile.get_values(my_task.name)
+        assert get_md5('result') == depfile.get_results(my_task.name)
```

Comment:
Indeed, it was working thanks to the typo on `task.results` :/
Now it fails because `self.dep_manager.get_results` returns the hash, so `save_success` saves the hash of the hash.